app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/provision_workflow/volumes.rb

```diff
@@ -0,0 +1,140 @@
+  rescue => e
+    logger(__method__).log_backtrace(e)
+  end
+
+  # Wait until zone is set, then fetch volumes that have a status of available and reside in that zone.
+  # @param _options [void]
+  # @return [Hash] Hash with ems_ref as key and name as value.
+  def cloud_volumes_to_volumes(_options = {})
+    zone = field(:placement_availability_zone)
+    return {} if zone.nil?
+
+    ar_volumes = ar_ems.cloud_volumes.select do |cloud_volume|
+      cloud_volume[:status] == 'available' && cloud_volume[:availability_zone_id] == zone
+    end
+    string_dropdown(ar_volumes)
+  rescue => e
+    logger(__method__).log_backtrace(e)
+  end
+
+  # Perform any needed manipulation and validation for new volume fields.
+  # @param values [Hash<Symbol, String>] Hash of all the keys and values submitted in the form.
+  # @return [void]
+  def parse_new_volumes_fields(values)
+    values[:new_volumes] = create_new_volumes_array(values)
+    validate_volumes(values[:new_volumes])
+  end
+
+  # Select keys that have one of the volume dialog keys followed by a digit.
+  # @param values [Hash<Symbol, String>] Hash of all the keys and values submitted in the form.
+  # @return [Array<Hash{Symbol => String, Integer}>]
+  def create_new_volumes_array(values)
+    new_volumes = []
+    # Regex to place the field name in the field index and count in the count index.
+    keys_regex = /(?<field>#{volume_dialog_keys.join('|')})_(?<count>\d+)/
+
+    values.each do |key, value|
+      next if value.nil? || (value.respond_to?(:length) && value.length.zero?)
+
+      reg = key.match(keys_regex)
+      next if reg.nil?
+
+      count = reg[:count].to_i
+      new_volumes[count] ||= {}
+
+      field = reg[:field].to_sym
+
+      # Use integer if value starts and ends with digits.
+      new_volumes[count][field] = value.match?('^\d+$') ? value.to_i : value
+    end
+
+    # The volume index starts at 1. Which means index 0 is nil. Compact to get rid of it.
+    new_volumes.compact
+  rescue => e
+    logger(__method__).log_backtrace(e, :context_msg => 'Creating new_volume array.')
+  end
+
+  # Validate the new volumes fields.
+  # @param new_volumes [Array<Hash{Symbol => String, Integer}>]
+  # @raise [StandardError] Validation of fields has failed. The error message contains a list of validation errors.
+  # @return [void]
+  def validate_volumes(new_volumes)
+    error_array = []
+    new_volumes.each_with_index do |item, zero_index|
+      index = zero_index + 1
+      validate_volume_required(index, item, error_array)
+      validate_volume_name(index, item[:volume_name], error_array)
+      validate_volume_size(index, item[:volume_size], error_array)
+      validate_volume_profile(index, item[:volume_profile], error_array)
+    end
+    e_msg = _("New volumes has the following problems: %{error}") % {:error => error_array.join(", ")}
+    raise e_msg unless error_array.length.zero?
```

Comment:
No need to build the error message if there is no error to raise

```suggestion
    unless error_array.empty?
      e_msg = _("New volumes has the following problems: %{error}") % {:error => error_array.join(", ")}
      raise e_msg 
    end
```

Additionally, prefer `.empty?` over `.length.zero?`  Alternatively, prefer avoiding a double negative (`unless ary.empty?`) with a positive (`if array.any?`)